### PR TITLE
ES3/ES5 emit support for computed properties

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Addresses #1840.

Probably best to go through the commits.

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -3601,6 +3750,7 @@ module ts {
             }
 
             function emitMemberAccessForPropertyName(memberName: DeclarationName) {
+                // TODO: (jfreeman,drosen): comment on why this is emitNode instead of emit here.
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: @JsonFreeman @sheetalkamat what is the appropriate behavior here?

File: src/compiler/emitter.ts [hunk 2]
```diff
@@ -3650,13 +3800,14 @@ module ts {
...
                         emitStart((<MethodDeclaration>member).name);
-                        emitNode(node.name);
+                        emitNode(node.name); // TODO (shkamat,drosen): comment for why emitNode instead of emit.
                         if (!(member.flags & NodeFlags.Static)) {
                             write(".prototype");
                         }
                         emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
                         emitEnd((<MethodDeclaration>member).name);
                         write(" = ");
+                        // TODO (drosen): Should we performing emitStart twice on emitStart(member)?
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: @sheetalkamat

File: src/compiler/emitter.ts (hunk 1 above)
Comment (JsonFreeman) [inline src/compiler/emitter.ts]: Probably emit

File: src/compiler/emitter.ts [hunk 3]
```diff
@@ -3676,6 +3925,7 @@ module ts {
                                 write(".prototype");
                             }
                             write(", ");
+                            // TODO: Shouldn't emitStart on name occur *here*?
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: @sheetalkamat?

**Pull request merged**